drivers/service_webhook.go

```diff
@@ -0,0 +1,120 @@
+package drivers
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	v1client "github.com/rancher/go-rancher/client"
+	"github.com/rancher/go-rancher/v2"
+
+	"github.com/rancher/webhook-service/config"
+	"github.com/rancher/webhook-service/model"
+)
+
+type ServiceWebhookDriver struct {
+}
+
+func (s *ServiceWebhookDriver) ValidatePayload(conf interface{}, apiClient *client.RancherClient) (int, error) {
+	if _, ok := conf.(model.ServiceWebhook); !ok {
+		return http.StatusInternalServerError, fmt.Errorf("Can't process config")
+	}
+	return http.StatusOK, nil
+}
+
+func (s *ServiceWebhookDriver) Execute(conf interface{}, apiClient *client.RancherClient, request *http.Request) (int, error) {
+	requestPayloadByte, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		return 500, err
+	}
+	rancherConfig := config.GetConfig()
+	webhookConfig := &model.ServiceWebhook{}
+	if err = mapstructure.Decode(conf, webhookConfig); err != nil {
+		return http.StatusInternalServerError, errors.Wrap(err, "Couldn't unmarshal config")
+	}
+
+	arry := strings.Split(request.RequestURI, "?")
+	projectID, errCode, err := getProjectID(request)
```

Comment:
By getting the projectID from the request URL,  I think you are introducing a  flaw where a user could change the the url and cause the webhook to be invoked on a service that they do not have access to. If you need the project ID, it needs to be in the webhook config.

Reply:
@cjellick I assume the webhook will be validated before it get into executing. It seems it doesn't, I will make the changes.